File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    ArrayList<Integer> list;
    public int sumNumbers(TreeNode root) {
        if(root==null)return 0;
        if(root.left == null && root.right ==null)return root.val;
        int numberDigitWithRoot = 0;
        list = new ArrayList();
        sumNumbersHelper(root, numberDigitWithRoot, list);
        int sum=0;
        for(Integer i: list)sum+=(int)i;
        return sum;
    }
    public void sumNumbersHelper(TreeNode root, int numberDigitWithRoot ,ArrayList<Integer> list){
        if(root == null)return;
        numberDigitWithRoot = numberDigitWithRoot*10 + root.val;
        if(root.left == null && root.right == null){
            list.add(numberDigitWithRoot);
        }
        sumNumbersHelper(root.left, numberDigitWithRoot, list);
        sumNumbersHelper(root.right, numberDigitWithRoot, list);
    }
}

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        total_sum = 0
        sums = []
        
        def dfs(root, number_formed):
            if not root:
                return
            
            new_number = number_formed * 10 + root.val
            if root.left is None and root.right is None:
                sums.append(new_number)
                
            dfs(root.left, new_number)
            dfs(root.right, new_number)
            

        dfs(root, 0)
        for i in sums:
            total_sum+=i
        return total_sum
        